client/src/main/scala/org/apache/celeborn/client/LifecycleManager.scala

```diff
@@ -497,43 +492,57 @@ class LifecycleManager(appId: String, val conf: CelebornConf) extends RpcEndpoin
       context: RpcCallContext,
       applicationId: String,
       shuffleId: Int,
-      mapId: Int,
-      attemptId: Int,
-      partitionId: Int,
-      oldEpoch: Int,
-      oldPartition: PartitionLocation,
-      cause: StatusCode): Unit = {
+      mapIds: util.List[Integer],
+      partitionIds: util.List[Integer],
+      oldEpochs: util.List[Integer],
+      oldPartitions: util.List[PartitionLocation],
+      causes: util.List[StatusCode]): Unit = {
+    val contextWrapper =
+      ChangeLocationsCallContext(context, partitionIds.size())
     // If shuffle not registered, reply ShuffleNotRegistered and return
     if (!registeredShuffle.contains(shuffleId)) {
       logError(s"[handleRevive] shuffle $shuffleId not registered!")
-      context.reply(ChangeLocationResponse(StatusCode.SHUFFLE_NOT_REGISTERED, None))
+      0 until partitionIds.size() foreach (idx => {
```

Comment:
There is no need to loop here I think here needs a bypass method.

Reply:
What does "bypass method" mean?